### PR TITLE
Add rules checking that builtin and custom libraries are imported in alphabetical order.

### PR DESCRIPTION
**Pull request by szymonslodkowski**
Add rules if builtin and custom libraries are sorted alphabetically

### CODE REVIEW

**Comment by bhirsz** [comment]
Thanks for the good PR. In a way it implements https://github.com/MarketSquare/robotframework-robocop/issues/107 . 
Though there is caveat here: order of the custom imports may be important (and it often is. For example you need to import B before you import A because B keeps variables you need for A). However it should be fine for built in imports. So the solution could be moving those rules (or at least the custom imports one) to community rules (which are disabled by default and can be enabled). What do you think about it?

File: robocop/checkers/misc.py [hunk 1]
```diff
@@ -586,6 +586,35 @@
         """,
         added_in_version="4.0.0",
     ),
+    "0926": Rule(
+        rule_id="0926",
+        name="builtin-imports-not-sorted",
+        msg="BuiltIn library import '{{ builtin_import }}' should be placed before '{{ previous_builtin_import }}'",
+        severity=RuleSeverity.WARNING,
```
Comment (bhirsz) [inline robocop/checkers/misc.py:593]:
```suggestion
        severity=RuleSeverity.WARNING,
        added_in_version="5.2.0",
```

File: robocop/checkers/misc.py [hunk 2]
```diff
@@ -586,6 +586,35 @@
...
+            Library    OperatingSystem
+            Library    Collections  # BuiltIn libraries imported not in alphabetical order
+
+        """,
+    ),
+    "0927": Rule(
+        rule_id="0927",
+        name="custom-imports-not-sorted",
+        msg="Custom library import '{{ custom_import }}' should be placed before '{{ previous_custom_import }}'",
+        severity=RuleSeverity.WARNING,
```
Comment (bhirsz) [inline robocop/checkers/misc.py:607]:
```suggestion
        severity=RuleSeverity.WARNING,
        added_in_version="5.2.0",
```

**Comment by szymonslodkowski** [comment]
@bhirsz thanks for quick review.
So I will leave the rule for builtins in the file I already implemented.
I was not aware about use cases in which order of imports is important - good to know that there are some. So  I will move rule checking rest of imports to custom rules.
Maybe there is also sense to implement rule verifying order of resources? I could try also to add such rule in this PR. Is it ok?

**Comment by bhirsz** [comment]
Yeah, feel free to do so - since it will be also custom rule then we can have such rule for resources as well (people can opt in for it).

**Comment by szymonslodkowski** [comment]
Ok, but I have a question about community rules numbering.
In robocop/checkers/community_rules there is only keywords.py with `RULE_CATEGORY_ID = "00"` I guess these new rules doesn't fit to this file. I would like to create new one, but what should be RULE_CATEGORY_ID and what should be rule_id?

**Comment by bhirsz** [comment]
You can decide it. But since we started with 00, then next one will be 01. Community rules are prefixed with 1 so final rule will be for example 10101, 10102...

File: robocop/checkers/misc.py (hunk 1 above)
Comment (szymonslodkowski) [reply robocop/checkers/misc.py:593]: Added

File: robocop/checkers/misc.py (hunk 2 above)
Comment (szymonslodkowski) [reply robocop/checkers/misc.py]: Added

**Review by bhirsz — APPROVED** [review]

**Pull request merged**